### ISSUE TITLE
Inserting line break causes caret to jump to beginning

### ISSUE THREAD

**Issue by philippfromme** [bpmn-io/dmn-js#297]
![2018-02-05_16-03-19](https://user-images.githubusercontent.com/7633572/35811184-20063572-0a8e-11e8-901f-2e1b61efff10.gif)

**Comment by nikku** [comment]
Unfortunately it turns out that this is not a proper fix, as it prevents updating of the component if other properties than `text` change.

To understand this, consider the following component usage:

```
<ContentEditable 
  text={ foo } 
  className={ foo.length > 3 ? 'long-text' : 'short-text' } />
```

`ContentEditable` will not update on `className` changes.


We could either (a) remove all other props from content editable (b) make it throw if the other props change or (c) implement proper selection updating on `text` changes.

**Comment by nikku** [comment]
A proper fix must be implemented.

**Comment by nikku** [comment]
Proper fix implemented via #309.
